src/Dofs/sparsity_pattern.jl

```diff
@@ -1,83 +1,473 @@
 
-See the [Sparsity Pattern](@ref) section of the manual.
+Instantiate a sparse matrix of type `Symmetric{Tv, SparseMatrixCSC{Tv, Ti}}`, i.e. a
+`LinearAlgebra.Symmetric`-wrapped `SparseMatrixCSC`, from the sparsity pattern `sp`. The
+resulting matrix will only store entries above, and including, the diagonal.
 """
-function create_symmetric_sparsity_pattern(dh::AbstractDofHandler; coupling=nothing,
-    topology::Union{Nothing, AbstractTopology} = nothing, cross_coupling = nothing)
-    return Symmetric(_create_sparsity_pattern(dh, nothing, true, true, coupling, topology, cross_coupling), :U)
+function allocate_matrix(::Type{Symmetric{Tv, S}}, sp::AbstractSparsityPattern) where {Tv, Ti, S <: SparseMatrixCSC{Tv, Ti}}
+    return Symmetric(_allocate_matrix(S, sp, #=sym=# true))
 end
 
 """
-    create_symmetric_sparsity_pattern(dh::AbstractDofHandler, ch::ConstraintHandler; coupling, topology::Union{Nothing, AbstractTopology}, cross_coupling)
+    allocate_matrix(sp::SparsityPattern)
+
+Allocate a sparse matrix of type `SparseMatrixCSC{Float64, Int}` from the sparsity pattern
+`sp`.
 
-Create a symmetric sparsity pattern accounting for affine constraints in `ch`. See
-the Affine Constraints section of the manual for further details.
+This method is a shorthand for the equivalent
+[`allocate_matrix(SparseMatrixCSC{Float64, Int}, sp)`]
+(@ref allocate_matrix(::Type{S}, sp::AbstractSparsityPattern) where {Tv, Ti, S <: SparseMatrixCSC{Tv, Ti}}).
 """
-function create_symmetric_sparsity_pattern(dh::AbstractDofHandler, ch::ConstraintHandler;
-        keep_constrained::Bool=true, coupling=nothing, topology::Union{Nothing, AbstractTopology} = nothing,
-        cross_coupling = nothing)
-    return Symmetric(_create_sparsity_pattern(dh, ch, true, keep_constrained, coupling, topology, cross_coupling), :U)
+allocate_matrix(sp::SparsityPattern) = allocate_matrix(SparseMatrixCSC{Float64, Int}, sp)
+
+"""
+    allocate_matrix(MatrixType, dh::DofHandler, args...; kwargs...)
+
+Allocate a matrix of type `MatrixType` from the DofHandler `dh`.
+
+This is a convenience method and is equivalent to:
+
+```julia
+sp = init_sparsity_pattern(dh)
+create_sparsity_pattern!(sp, dh, args...; kwargs...)
+allocate_matrix(MatrixType, sp)
+````
+
+Refer to [`allocate_matrix`](@ref allocate_matrix(::Type{<:Any}, ::SparsityPattern)) for
+supported matrix types, and to [`create_sparsity_pattern`](@ref) for details about supported
+arguments `args` and keyword arguments `kwargs`.
+
+!!! note
+    If more than one sparse matrix is needed (e.g. a stiffness and a mass matrix) it is more
+    efficient to explicitly create the sparsity pattern instead of using this method, i.e.
+    use
+    ```julia
+    sp = init_sparsity_pattern(dh)
+    create_sparsity_pattern!(sp, dh)
+    K = allocate_matrix(sp)
+    M = allocate_matrix(sp)
+    ```
+    instead of
+    ```julia
+    K = allocate_matrix(dh)
+    M = allocate_matrix(dh)
```

Comment:
What about
```julia
    K = allocate_matrix(dh)
    M = deepcopy(K)
```
?

Reply:
Deepcopy should pretty much never be used.

Reply:
Why? Don't we want a fully recursive copy of the sparse matrix with a duplication of the internal arrays here?

Reply:
`copy` is enought here and would work for perfectly fine for `SparseMatrixCSC` but doesn't work in general (e.g. no copy for HYPREMatrix etc). I can add a note.

Reply:
> Deepcopy should pretty much never be used.

@fredrikekre has told me the same, but is this documented somewhere in the julia ecosystem?

Reply:
> Why?

See e.g. https://github.com/JuliaLang/julia/issues/42796#issuecomment-951232853